[CS2103T-F11-1] Hackathon HR Manager

hcwong opened the pull request.

hcwong committed:
Initial commit (#1)

hcwong committed:
Add PR template (#8)

brianyenna committed:
Morph storage (#11)

* Update build.gradle file (minor changes)

* Complete set-up for project (as per the Developer Guide)

* Update README.adoc with Travis CI build status

* Update some files to ensure that ./config/travis/run-checks.sh runs fine

* Create the 4 different ListStorages. Create AlfredStorage. Create AlfredStorageManager (halfway done).

* Creation of all necessary Storage classes (with the exception of the TeamListStorage handler classes)

* Added Storage classes for TeamList

* Fix bracket issue

justarock111 committed:
Change Exception and UI class

justarock111 committed:
Revert "Change Exception and UI class" (#16)

This reverts commit 6e60a08e38ad76f4c6ad054c54ef605bb195e123.

justarock111 committed:
Final changes to Feat/change exception class (#17)

* Exception interface updates

* Exception interface updates v1.1

* add parenthesis

* Update DataConversionException.java

* Update ParseException.java

* Update DuplicateEntityException.java

* Update Messages.java

hcwong committed:
Implement Model public methods (#7)

* Implemented Entity abstract class

* Add EntityList Interface

* Add method to check for existence

* Refactor EntityListInterface

* Demo flow of how controller will call model via modelManager

* Add team object

* Create mentor class

* Add issue type

* Add the 4 lists. Now they are fairly repetitive, but do not want to prematurely optimize

* Modify the Model API to reflect latest changes

* Rename ID to Id and expose viewMinimal and viewDetailed methods

* Add equals, toString, hash for all

* Add line

hcwong committed:
Refactor/model api (#19)

* Move all the CRUD methods from EntityList to ModelManager

* Remove the override methods from the various EntityLists as they are now superflous

* Remove circular references from entity objects

* Created ReadableEntityList to abstract and implemented Participant methods

* Implement API for all 4 entities. Pending only saving and load of model

* Standardize all entities to have only 1 constructor as per Brian's comment

* Make model throw AlfredRuntimeException

* Rename package to use lowercase

* Uncomment old person type methods to reduce errors

brianyenna committed:
Morph storage (#18)

* Update build.gradle file (minor changes)

* Complete set-up for project (as per the Developer Guide)

* Update README.adoc with Travis CI build status

* Update some files to ensure that ./config/travis/run-checks.sh runs fine

* Create the 4 different ListStorages. Create AlfredStorage. Create AlfredStorageManager (halfway done).

* Creation of all necessary Storage classes (with the exception of the TeamListStorage handler classes)

* Added Storage classes for TeamList

* Fix bracket issue

* Update Storage classes with appropriate JSON adaptation

john0227 committed:
Implemented Participant Commands (#14)

* Made changes to fit team's scope.

* Added AddCommands and DeleteCommands.

* Added Edit Commands

* Added List and View commands.

* Corrected repoUrl

* Made changes to follow Joshua's implementations. Contains vaguely-implemented methods.

* Changed parameters from EntityList to Model. Made some changes to follow Joshua's implementations.

* Moved ENTITY_TYPE to a static constant stored in CliSyntax.

* Updated code to use viewMinimal() and viewDetailed().

* Minor changes

* Implemented potential ways of executing each Participant command.

* Removed Commands associated with Issues.

(cherry picked from commit cef69759ba683ab153084b6e274ea11472a12365)

john0227 committed:
Updated EditParticipantCommand

Updated code to use return value of editParticipant()

justarock111 committed:
Feat/validate objects (#15)

* Added object validation

* Edited code after review

john0227 committed:
Imported Entities and other objects (#2)

* Made changes to fit team's scope.

* Added AddCommands and DeleteCommands.

* Added Edit Commands

* Added List and View commands.

* Corrected repoUrl

* Made changes to follow Joshua's implementations. Contains vaguely-implemented methods.

* Changed parameters from EntityList to Model. Made some changes to follow Joshua's implementations.

* Moved ENTITY_TYPE to a static constant stored in CliSyntax.

* Updated code to use viewMinimal() and viewDetailed().

* Minor changes

* Implemented potential ways of executing each Participant command.

* Merge from upstream

* Updated code to use Model#updateParticipant's return value.

* Removed Commands associated with Issues.

* Removed Commands associated with Issues.

(cherry picked from commit cef69759ba683ab153084b6e274ea11472a12365)

* Imported Entities and other objects.

brianyenna committed:
Update user guide (#26)

* Update build.gradle file (minor changes)

* Complete set-up for project (as per the Developer Guide)

* Update README.adoc with Travis CI build status

* Update some files to ensure that ./config/travis/run-checks.sh runs fine

* Create the 4 different ListStorages. Create AlfredStorage. Create AlfredStorageManager (halfway done).

* Creation of all necessary Storage classes (with the exception of the TeamListStorage handler classes)

* Added Storage classes for TeamList

* Fix bracket issue

* Update Storage classes with appropriate JSON adaptation

* Update User Guide

* Update README.adoc

brianyenna committed:
Update user guide (#26)

* Update build.gradle file (minor changes)

* Complete set-up for project (as per the Developer Guide)

* Update README.adoc with Travis CI build status

* Update some files to ensure that ./config/travis/run-checks.sh runs fine

* Create the 4 different ListStorages. Create AlfredStorage. Create AlfredStorageManager (halfway done).

* Creation of all necessary Storage classes (with the exception of the TeamListStorage handler classes)

* Added Storage classes for TeamList

* Fix bracket issue

* Update Storage classes with appropriate JSON adaptation

* Update User Guide

* Update README.adoc

john0227 committed:
Changes to AboutUs.adoc and ContactUs.adoc (#29)

* Made changes to fit team's scope.

* Added AddCommands and DeleteCommands.

* Added Edit Commands

* Added List and View commands.

* Corrected repoUrl

* Made changes to follow Joshua's implementations. Contains vaguely-implemented methods.

* Changed parameters from EntityList to Model. Made some changes to follow Joshua's implementations.

* Moved ENTITY_TYPE to a static constant stored in CliSyntax.

* Updated code to use viewMinimal() and viewDetailed().

* Minor changes

* Implemented potential ways of executing each Participant command.

* Merge from upstream

* Updated code to use Model#updateParticipant's return value.

* Removed Commands associated with Issues.

* Removed Commands associated with Issues.

(cherry picked from commit cef69759ba683ab153084b6e274ea11472a12365)

* Imported Entities and other objects.

* Updated AboutUs.adoc and ContactUs.adoc

Abhiman2211 committed:
Update UserGuide Introduction and Quickstart

Abhiman2211 committed:
Update UserGuide Section 4

hcwong committed:
Move generate id to readable entityList so that controller and parser can call it (#31)

Abhiman2211 committed:
Alfred Add Command Parsers (#32)

* Add Alfred-specific parser classes.

* Modify prefixes.

* Implement Add Command Parsers.

hcwong committed:
Fix checkStyle errors

hcwong committed:
Retrigger travis build

john0227 committed:
Made corrections to teammate names (#35)

hcwong committed:
Restart for travis gh-pages

john0227 committed:
Delete Abhiman2211.png

john0227 committed:
Delete joanna-YJA.png

john0227 committed:
Added teammate pictures (#36)

hcwong committed:
Make changes to developer guide for week 7 milestone (#25)

justarock111 committed:
Change developer guide (#38)

hcwong committed:
Fix typos on DG (#39)

john0227 committed:
Update README.adoc to show Netlify status

john0227 committed:
Corrected Netlify status badge snippet

john0227 committed:
Update README.adoc

john0227 committed:
Update README.adoc

marcusteh1238 commented on docs/DeveloperGuide.adoc:
```diff
@@ -279,51 +287,88 @@ Refer to the guide <<DevOps#, here>>.
-(For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
+(For all use cases below, the *System* is the `HackathonManager` and the *Actor* is the `user`, unless specified otherwise)
 
 [discrete]
```
This should be numbered together with the rest of Use Cases and should not be discrete.

marcusteh1238 commented on docs/DeveloperGuide.adoc:
```diff
@@ -334,23 +379,112 @@ Use case ends.
+
+*MSS*
+
+1.  User requests a find an Entity of a specific Entity Type.
```
Grammar error. Should be "User requests `to` find ..."

marcusteh1238 commented on docs/DeveloperGuide.adoc:
```diff
@@ -334,23 +379,112 @@ Use case ends.
++
+Use case ends.
+
+*Extensions*
```
Should you add an extension for invalid index? I see in C3 that you can interact with the list through both index and name.

marcusteh1238 commented on docs/DeveloperGuide.adoc:
```diff
@@ -334,23 +379,112 @@ Use case ends.
++
+Use case ends.
+
+=== Use case: Create an Entity of a specific Entity Type(Participant, Mentor, Team)
```
Should contain an extension for when the entity type specified is invalid.

marcusteh1238 commented on docs/DeveloperGuide.adoc:
```diff
@@ -334,23 +379,112 @@ Use case ends.
++
+Use case resumes from step 4.
+
+=== Use case: Find the top scoring Teams for each category
```
This use case shows up twice in the developer guide.

marcusteh1238 commented on docs/DeveloperGuide.adoc:
```diff
@@ -279,51 +287,88 @@ Refer to the guide <<DevOps#, here>>.
+
+|`* * *` |Admin In-Charge |keep track of winning teams and the prizes won | ensure that the prize-giving ceremony runs smoothly
+
+|`* * *` |Admin In-Charge |make sure that I will be notified on any wrong commands that I type | make sure that I do not accidentally clutter up my list of entries
```
If I understood correctly, the notification for wrong commands should be "so that I can re-enter the correct command".

marcusteh1238 commented on docs/DeveloperGuide.adoc:
```diff
@@ -279,51 +287,88 @@ Refer to the guide <<DevOps#, here>>.
+
+|`* *` |Admin In-Charge |know my sponsor's needs and arrival time | adequately cater to their needs and allocate manpower accordingly
+
+|`* *` |Admin In-Charge |keep track of inventory of swag | make sure they are adequately catered to all participants
```
What does swag refer to here? If possible, add it into the glossary for readers to understand. Even though it is in the user guide, it has no definition.

brianyenna committed:
Fix Storage Compilation Errors (#42)

* Update build.gradle file (minor changes)

* Complete set-up for project (as per the Developer Guide)

* Update README.adoc with Travis CI build status

* Update some files to ensure that ./config/travis/run-checks.sh runs fine

* Create the 4 different ListStorages. Create AlfredStorage. Create AlfredStorageManager (halfway done).

* Creation of all necessary Storage classes (with the exception of the TeamListStorage handler classes)

* Added Storage classes for TeamList

* Fix bracket issue

* Update Storage classes with appropriate JSON adaptation

* Update User Guide

* Fixed all compilation problems

hcwong committed:
Update gitignore to ignore binary files (#44)

brianyenna committed:
Fix compileTestJava (#45)

* Update build.gradle file (minor changes)

* Complete set-up for project (as per the Developer Guide)

* Update README.adoc with Travis CI build status

* Update some files to ensure that ./config/travis/run-checks.sh runs fine

* Create the 4 different ListStorages. Create AlfredStorage. Create AlfredStorageManager (halfway done).

* Creation of all necessary Storage classes (with the exception of the TeamListStorage handler classes)

* Added Storage classes for TeamList

* Fix bracket issue

* Update Storage classes with appropriate JSON adaptation

* Update User Guide

* Fixed all compilation problems

* Stop-gap fix for compileTestJava

* Stop-gap measures to fix compileTestJava while fixing codebase

Abhiman2211 committed:
Fix add participant parser (#47)

john0227 committed:
Checkstyle: Commands and Parsers (#51)

john0227 committed:
Checkstyle EntityList (#53)

john0227 committed:
Checkstyle: SubjectName.java (#54)

justarock111 committed:
Feat/checkstyle (#55)

justarock111 committed:
Feat/checkstyle (#56)

brianyenna committed:
Implement Basic Storage Tests and Fix Compile Errors (#57)

* Update build.gradle file (minor changes)

* Complete set-up for project (as per the Developer Guide)

* Update README.adoc with Travis CI build status

* Update some files to ensure that ./config/travis/run-checks.sh runs fine

* Create the 4 different ListStorages. Create AlfredStorage. Create AlfredStorageManager (halfway done).

* Creation of all necessary Storage classes (with the exception of the TeamListStorage handler classes)

* Added Storage classes for TeamList

* Fix bracket issue

* Update Storage classes with appropriate JSON adaptation

* Update User Guide

* Fixed all compilation problems

* Stop-gap fix for compileTestJava

* Stop-gap measures to fix compileTestJava while fixing codebase

* Ensure that ParticipantList can be saved and read from file

* Remove all dependencies on Issues and IssueList for Storage

* Create TypicalMentors.java file

* Update SubjectName enum values

* Update TypicalMentors.java

* Fix compileTestJava compilation errors

hcwong committed:
Fix remaining checkstyle errors (#58)

hcwong committed:
Chore/skip tests (#48)

hcwong committed:
Merge pull request #59 from AY1920S1-CS2103T-F11-1/merge-develop

Merge develop branch into master

hcwong committed:
Replace the == with .equals (#60)

Abhiman2211 committed:
Edit add command parsers (#62)

hcwong committed:
Task/update exceptions (#52)

* Move exceptions

* Update Exceptions

* Fix conflicts after rebase

john0227 committed:
MentorCommands and TeamCommands (#63)

brianyenna committed:
Update storage (#78)

hcwong committed:
Add model tests and add cross updating when CRUD operation (#61)

hcwong committed:
Make generateId static and move model initialization to modelManager (#65)

hcwong committed:
Hotfix to expose FilteredList for find command (#88)

justarock111 committed:
Add entity card and entity panel (#89)

hcwong committed:
Customize error message (#82)

hcwong committed:
Implement looser definition of equality (#85)

john0227 committed:
Implemented tests for commands (#79)

hcwong committed:
Implement find command (#86)

* Implement find function on model

* Add tests

* Implement commands

* Implement parser for find command

* Rename parsers and add it into alfredparser

* Make message usage clearer

* Fix typo

Abhiman2211 committed:
Add Alfred Parsers (#98)

brianyenna committed:
Implement model history (#104)

* Create HistoryRecord and HistoryManager. Update entities and objects within entities with deep copy() method

* Update all Commands with a call to Model#updateHistory()

* Update javadoc for ModelManager#updateHistory() method

* Update constructor for ModelHistoryRecord with deep copies of the EntityLists

* Fix checkstyle and update ModelManagerStub with placeholder updateHistory() method

* Update the exceptions thrown in the EntityList classes

* Update javadocs

brianyenna committed:
Implement model history tests (#108)

* Create HistoryRecord and HistoryManager. Update entities and objects within entities with deep copy() method

* Update all Commands with a call to Model#updateHistory()

* Update javadoc for ModelManager#updateHistory() method

* Update constructor for ModelHistoryRecord with deep copies of the EntityLists

* Fix checkstyle and update ModelManagerStub with placeholder updateHistory() method

* Create tets for ModelHistoryManager

* Update ModelHistoryManager tests (for undo command)

* Update MentorList.java. Fix minor problem/artifact from git merge

brianyenna committed:
Implement undo command (#107)

* Create HistoryRecord and HistoryManager. Update entities and objects within entities with deep copy() method

* Update all Commands with a call to Model#updateHistory()

* Update javadoc for ModelManager#updateHistory() method

* Update constructor for ModelHistoryRecord with deep copies of the EntityLists

* Fix checkstyle and update ModelManagerStub with placeholder updateHistory() method

* Create tets for ModelHistoryManager

* Update ModelHistoryManager tests (for undo command)

* Create UndoCommand class

* Fix checkstyle violations

* Added undo command command word to Alfred Parser. Effects are now visible on GUI

* Fix checkstyle error

justarock111 committed:
Feat/display entity list (#100)

* Changed files to allow user interface ti display proper content during initialisation

* Fix checkstyle

* Update MESSAGE_USAGE of commands

* update Massage Usag for command

* Update UI to show CommandResult

* Fix checkstyle

* Morph User Interface to display entity list

* Change Entity list displayed for commands

* Fix bug

* Fix compile error

* Fix compile error

* Update MentorList.java

* Fix checkstyle

* Fix checkstyle

* Update AddMentorCommand.java

* Fix checkstyle

* Fix checkstyle

* Fix checkstyle

* Fix bug

* Fix checkstyle

brianyenna committed:
Fix storage bugs (#123)

* Fix bug causing addTeam command to fail. Issue has been identified to be the saving of Optional<Mentor> object in TeamList

* Remove debug statements

* Remove debug statements

* Update all Storage files with more cleaner exception throwing and handling

* Change comment typo

brianyenna committed:
Fix problem of default values populating teamlist.json and write edge test cases (#124)

* Fix bug causing addTeam command to fail. Issue has been identified to be the saving of Optional<Mentor> object in TeamList

* Remove debug statements

* Remove debug statements

* Update all Storage files with more cleaner exception throwing and handling

* Change comment typo

* Fixed JsonTeamListStorageTest bug that causes teamlist.json to be populated with default values everytime a test is run

* Update JsonTeamListStorage with tests for Optional<Mentor> and empty ParticipantList

* Refactor JsonTeamListStorageTest and update javadocs

john0227 committed:
Updates on AlfredParserUtil (#116)

john0227 committed:
Update ProjectType.java

hcwong committed:
Fixes the ID generation of entity objects (#125)

Abhiman2211 committed:
Update UserGuide.adoc

Abhiman2211 committed:
Enhance Standardisation of Commands (#127)

Abhiman2211 committed:
Rename Alfred parser util methods (#130)

Abhiman2211 committed:
Update User Guide

Add guide to leaderboard and scoring features.

john0227 committed:
Feat/bulk registration (#87)

hcwong committed:
Updated User and Developer Guide (#145)

Abhiman2211 committed:
Userguide (#147)

john0227 committed:
Updated User Guide to reflect Import/Export Commands (#144)

brianyenna committed:
Standardise User Guide and Update User Guide with Undo/Redo Command Information (#150)

justarock111 committed:
Update user guide (#149)

justarock111 committed:
Update developer guide (#153)

brianyenna committed:
Update developer guide (Storage) (#152)

* Create AlfredStorageClassDiagram .png and .puml files

* Completed Storage section of Developer Guide

Abhiman2211 committed:
Developerguide (#154)

brianyenna committed:
Update DeveloperGuide.adoc

Remove addressbook references

brianyenna committed:
Update DeveloperGuide.adoc

Update comments

brianyenna committed:
Implement Command History and Update ModelHistoryManager (#143)

brianyenna committed:
Update ID.java regex and solve empty EntityList in JSON bug (#158)

justarock111 committed:
Redesign User Interface (#159)

Abhiman2211 committed:
Remove oldparsers (#161)

hcwong committed:
Fix/reset filteredlist (#164)

Kyzure commented:
• The XYZ and ABC command explanation criss-cross and are very far away from the labels. Consider moving the explanation to the left side of the diagram.
• `//// end::undoredo[]` was found in the Proposed Undo Redo Feature Section.
• Regarding the BetterModelClassDiagram.png, consider giving mentor name multiplicity of 2 instead of having 2 separate arrows indicating 1 name. The diagram can also be abstracted out/split into smaller diagrams to better clarity and readability. Either that or added labels to indicate what each arrow stands for.
• AlfredStorageClassDiagram.png is well written and drawn.
• There is a lack of activity diagrams used.
• Certain class diagrams lack the labelling of their package names.

shiyao821 commented on docs/DeveloperGuide.adoc:
```diff
@@ -62,7 +71,6 @@ For example, the `Logic` component (see the class diagram given below) defines i
 image::LogicClassDiagram.png[]
```
![Screenshot 2019-10-25 at 1 18 27 PM](https://user-images.githubusercontent.com/42567064/67545026-30beab80-f72a-11e9-962a-8f49949204a3.png)

Words are too small -  not unreadable
May be cluttered with too many unnecessary details
Should give explanation of how it works

shiyao821 replied:
![Screenshot 2019-10-25 at 1 18 27 PM](https://user-images.githubusercontent.com/42567064/67545107-8a26da80-f72a-11e9-8c32-165c531d3762.png)

Words are too small -  not unreadable
May be cluttered with too many unnecessary details
White space can be better used - maximise empty spaces

shiyao821 commented on docs/DeveloperGuide.adoc:
```diff
@@ -75,21 +83,27 @@ The sections below give more details of each component.
+[NOTE]
+UI image and description to be updated with the later milestones.
+
 image::UiClassDiagram.png[]
```
![Screenshot 2019-10-25 at 1 26 12 PM](https://user-images.githubusercontent.com/42567064/67545265-1afdb600-f72b-11e9-8947-4ee980194a04.png)
Lacks Figure Label. Format can be made consistent as with the previous diagram. 

Should TeamListPanel, MentorListPanel etc be under the UI box?

shiyao821 commented on docs/DeveloperGuide.adoc:
```diff
@@ -75,21 +83,27 @@ The sections below give more details of each component.
 
 * Executes user commands using the `Logic` component.
-* Listens for changes to `Model` data so that the UI can be updated with the modified data.
+* Looks at the prefix type given in the CommandResult(returned after every Command is executed), as displays the specific EntityListPanel respectively.
```
EntityListPanel could be in `code` format for readability

shiyao821 commented on docs/DeveloperGuide.adoc:
```diff
@@ -98,56 +112,80 @@ image::LogicClassDiagram.png[]
 
 .Structure of the Model Component
-image::ModelClassDiagram.png[]
+image::BetterModelClassDiagram.png[]
```
Diagram could be made less cluttered (and perhaps shorter). Diagrams with many arrows crossing each other are not readable

Name Class at the bottom left has 2 arrows coming in from Mentor, each with multiplicity of 1. Could be fused to 1 array with a multiplicity of 2.

ning-sy210 replied:
Diagram is too cluttered.

shiyao821 commented on docs/DeveloperGuide.adoc:
```diff
@@ -98,56 +112,80 @@ image::LogicClassDiagram.png[]
+, 3 `FilteredLists` and a `UserPrefs` Object and does all the validation logic needed for the application
+
+.Simple Illustration of ModelManager
+image::ModelManagerClassDiagram.png[]
```
Simple and neat

ning-sy210 replied:
The Class Diagrams here are very simple and neat, which makes them easy to understand.

shiyao821 commented on docs/DeveloperGuide.adoc:
```diff
@@ -98,56 +112,80 @@ image::LogicClassDiagram.png[]
+.Simple Illustration of ModelManager
+image::ModelManagerClassDiagram.png[]
+
+.EntityList simplified structure
```
Lacks explanation for now

shiyao821 commented on docs/DeveloperGuide.adoc:
```diff
@@ -98,56 +112,80 @@ image::LogicClassDiagram.png[]
 
 .Structure of the Storage Component
-image::StorageClassDiagram.png[]
+image::AlfredStorageClassDiagram.png[]
```
Neatly done

ning-sy210 replied:
The Class Diagram here is very neatly organized and well-illustrated.

shiyao821 commented on docs/DeveloperGuide.adoc:
```diff
@@ -157,86 +195,94 @@ This section describes some noteworthy details on how certain features are imple
+//* **Alternative 2:** Use `HistoryManager` for undo/redo
+//** Pros: We do not need to maintain a separate list, and just reuse what is already in the codebase.
+//** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates //Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things.
+//// end::undoredo[]
```
Can be commented out. Currently showing in Netlify

justarock111 committed:
Redesign User Interface to display HistoryCommand (#167)

ning-sy210 commented on docs/DeveloperGuide.adoc:
```diff
@@ -75,21 +83,27 @@ The sections below give more details of each component.
+Logic Architecture image and description to be updated with the later milestones.
 
 [[fig-LogicClassDiagram]]
 .Structure of the Logic Component
```
The Architecture Diagram here contains words that are too small to see. Try breaking it up into separate diagrams to improve readability.

hcwong committed:
Implement find by multiple fields (#128)

hcwong committed:
Comma separate the entities when they are printed out (#156)

john0227 committed:
Updated ViewCommand, Refactored ListCommands, Other minor updates (#168)

brianyenna committed:
Revert "Updated ViewCommand, Refactored ListCommands, Other minor updates (#168)" (#171)

This reverts commit c44c907aee7fa403b0f3639d73bc054baa913feb.

brianyenna committed:
Update Undo Feature (#173)

brianyenna committed:
Implement redo (#185)

hcwong committed:
Update Help Command (#170)

hcwong committed:
Deprecate Address Book (#160)

john0227 committed:
Updated parsertests branch with master branch (#192)

Abhiman2211 committed:
Feature judging (#135)

john0227 committed:
Updated ViewCommand, Refactored ListCommands, Other minor updates (#180)

john0227 committed:
Creates an error file after ImportCommand (#182)

john0227 committed:
Docs/user guide (#199)

brianyenna committed:
Implement command navigation (#190)

* Preliminary code changes to ensure that TextField displays properly

* Implement basic command navigation feature

* Changed CommandHistoryManager implementation from 2 stacks to 1 LinkedList

* Remove debug statements

hcwong committed:
Update find UG (#200)

brianyenna committed:
Update user guide v1.3 (#202)

* Update javadocs for setCommandNavigationHandler()

* Update User Guide formating. Delete commandType duplicate error. Update user guide for history, undo, redo and command navigation

Abhiman2211 committed:
Feature leaderboard (#134)

Abhiman2211 committed:
Leaderboard userguide (#198)

hcwong committed:
Add find tests (#187)

brianyenna committed:
Update UserGuide 

Minor corrections for `history` command section. 

Update Command Summary section to indicate features that come in v2.0

brianyenna committed:
Update UserGuide.adoc

Update with further corrections for `history` command section

brianyenna committed:
Update UserGuide.adoc (`history` command section)

brianyenna committed:
Update UserGuide.adoc

Updated `command navigation` section

brianyenna committed:
Update user guide v1.3 (#209)

* Update javadocs for setCommandNavigationHandler()

* Update User Guide formating. Delete commandType duplicate error. Update user guide for history, undo, redo and command navigation

* Fix userguide

john0227 committed:
Made changes to documentation (#206)

justarock111 committed:
Implement Assign Command (#196)

hcwong committed:
Update release version (#212)

john0227 committed:
Fixed issues with deleting (#207)

justarock111 committed:
Update User guide (#215)

justarock111 committed:
Fix typo

justarock111 committed:
fix bug (#217)

justarock111 committed:
Feat/fix (#218)

justarock111 committed:
Feat/update user guide v1.3 (#219)

brianyenna committed:
Change user guide 1 nov (#220)

* change user guide as per TA feedback

* Update User Guide

* Update QuickStart

Abhiman2211 committed:
Docs userguide (#216)

brianyenna committed:
Fix command navigation (#222)

* Simple fix for command navigation feature

* Update Commands with TrackableState marker interface and model.updateHistory() and model.recordCommandExecution() in their respective execute() methods

Abhiman2211 committed:
Scoring test (#242)

Abhiman2211 committed:
Pe fixes (#275)

brianyenna committed:
Implement Multiple Undo/Redo Feature (#240)

* Simple fix for command navigation feature

* Update Commands with TrackableState marker interface and model.updateHistory() and model.recordCommandExecution() in their respective execute() methods

* Implement multiple undo-redo feature. Create tests. Display command input string in History section of UI

* Change panel colours for delimiters in History section

brianyenna committed:
Deprecate ab3 storage (#279)

* Deprecate AB3 storage code. Update HistoryCommand with parser to prevent abuse. Ensure that undo/redo saves to Storage

* Delete LoadException.java

* Deprecate all remaining storage classes

brianyenna committed:
Implement capacity of 50 for CommandHistoryManager (#241)

* Simple fix for command navigation feature

* Update Commands with TrackableState marker interface and model.updateHistory() and model.recordCommandExecution() in their respective execute() methods

* Implement multiple undo-redo feature. Create tests. Display command input string in History section of UI

* Change panel colours for delimiters in History section

* Implement capacity of 50 for CommandHistoryManager

* Changed capacity back to 50 commands

* Update capacity to CAPACITY

hcwong committed:
Implement negative find and find by union (#274)

brianyenna committed:
Change to use Alt Modifier for Command History Navigation Feature (#283)

* Change to alt+up/down keys for command history navigation. Fix index bug in feature

* Alt Key Combination

brianyenna committed:
Update developer guide v1.4 (#287)

Abhiman2211 committed:
Test leaderboard (#280)

justarock111 committed:
Final fixed command suggestion box (#281)

justarock111 committed:
Implement Homepage to show Statistics (#197)

justarock111 committed:
Feat/dark mode (#284)

Abhiman2211 committed:
Update README.adoc

Abhiman2211 committed:
Add Coveralls

Abhiman2211 committed:
Rectify Formatting

justarock111 committed:
Fix bug (#288)

justarock111 committed:
Update User Guide, Developer Guide and Personal Project Portfolio (#293)

Abhiman2211 committed:
Devguide (#292)

justarock111 committed:
Fix bug in command suggestion box and updates feature to accomodate new commands (#294)

justarock111 committed:
Fix bug (#296)

Abhiman2211 committed:
Remove Coveralls badge

justarock111 committed:
Set maximum team size (#297)

justarock111 committed:
Update color of CommandRecordCard and File Menu Item (#300)

hcwong committed:
Add entertainment and other into SubjectName (#289)

john0227 committed:
Resolved Issues raised during the PE (#291)

Abhiman2211 committed:
Refactorscore (#298)

Abhiman2211 committed:
user guide fix (#304)

brianyenna committed:
Fix undo redo ui response (#302)

hcwong committed:
Add codecov (#301)

justarock111 committed:
Update colors (#303)

Abhiman2211 committed:
Final improvement (#299)

justarock111 committed:
set minimum window size (#308)

john0227 committed:
EditCommand (#310)

* Merge from upstream

* Updated code to use Model#updateParticipant's return value.

* Change Exception and UI class

* Revert "Change Exception and UI class" (#16)

This reverts commit 6e60a08e38ad76f4c6ad054c54ef605bb195e123.

* Final changes to Feat/change exception class (#17)

* Exception interface updates

* Exception interface updates v1.1

* add parenthesis

* Update DataConversionException.java

* Update ParseException.java

* Update DuplicateEntityException.java

* Update Messages.java

* Implement Model public methods (#7)

* Implemented Entity abstract class

* Add EntityList Interface

* Add method to check for existence

* Refactor EntityListInterface

* Demo flow of how controller will call model via modelManager

* Add team object

* Create mentor class

* Add issue type

* Add the 4 lists. Now they are fairly repetitive, but do not want to prematurely optimize

* Modify the Model API to reflect latest changes

* Rename ID to Id and expose viewMinimal and viewDetailed methods

* Add equals, toString, hash for all

* Add line

* Refactor/model api (#19)

* Move all the CRUD methods from EntityList to ModelManager

* Remove the override methods from the various EntityLists as they are now superflous

* Remove circular references from entity objects

* Created ReadableEntityList to abstract and implemented Participant methods

* Implement API for all 4 entities. Pending only saving and load of model

* Standardize all entities to have only 1 constructor as per Brian's comment

* Make model throw AlfredRuntimeException

* Rename package to use lowercase

* Uncomment old person type methods to reduce errors

* Morph storage (#18)

* Update build.gradle file (minor changes)

* Complete set-up for project (as per the Developer Guide)

* Update README.adoc with Travis CI build status

* Update some files to ensure that ./config/travis/run-checks.sh runs fine

* Create the 4 different ListStorages. Create AlfredStorage. Create AlfredStorageManager (halfway done).

* Creation of all necessary Storage classes (with the exception of the TeamListStorage handler classes)

* Added Storage classes for TeamList

* Fix bracket issue

* Update Storage classes with appropriate JSON adaptation

* Removed Commands associated with Issues.

* Removed Commands associated with Issues.

(cherry picked from commit cef69759ba683ab153084b6e274ea11472a12365)

* Implemented Participant Commands (#14)

* Made changes to fit team's scope.

* Added AddCommands and DeleteCommands.

* Added Edit Commands

* Added List and View commands.

* Corrected repoUrl

* Made changes to follow Joshua's implementations. Contains vaguely-implemented methods.

* Changed parameters from EntityList to Model. Made some changes to follow Joshua's implementations.

* Moved ENTITY_TYPE to a static constant stored in CliSyntax.

* Updated code to use viewMinimal() and viewDetailed().

* Minor changes

* Implemented potential ways of executing each Participant command.

* Removed Commands associated with Issues.

(cherry picked from commit cef69759ba683ab153084b6e274ea11472a12365)

* Updated EditParticipantCommand

Updated code to use return value of editParticipant()

* Feat/validate objects (#15)

* Added object validation

* Edited code after review

* Imported Entities and other objects.

* Imported Entities and other objects (#2)

* Made changes to fit team's scope.

* Added AddCommands and DeleteCommands.

* Added Edit Commands

* Added List and View commands.

* Corrected repoUrl

* Made changes to follow Joshua's implementations. Contains vaguely-implemented methods.

* Changed parameters from EntityList to Model. Made some changes to follow Joshua's implementations.

* Moved ENTITY_TYPE to a static constant stored in CliSyntax.

* Updated code to use viewMinimal() and viewDetailed().

* Minor changes

* Implemented potential ways of executing each Participant command.

* Merge from upstream

* Updated code to use Model#updateParticipant's return value.

* Removed Commands associated with Issues.

* Removed Commands associated with Issues.

(cherry picked from commit cef69759ba683ab153084b6e274ea11472a12365)

* Imported Entities and other objects.

* Updated AboutUs.adoc and ContactUs.adoc

* Update user guide (#26)

* Update build.gradle file (minor changes)

* Complete set-up for project (as per the Developer Guide)

* Update README.adoc with Travis CI build status

* Update some files to ensure that ./config/travis/run-checks.sh runs fine

* Create the 4 different ListStorages. Create AlfredStorage. Create AlfredStorageManager (halfway done).

* Creation of all necessary Storage classes (with the exception of the TeamListStorage handler classes)

* Added Storage classes for TeamList

* Fix bracket issue

* Update Storage classes with appropriate JSON adaptation

* Update User Guide

* Update README.adoc

* Changes to AboutUs.adoc and ContactUs.adoc (#29)

* Made changes to fit team's scope.

* Added AddCommands and DeleteCommands.

* Added Edit Commands

* Added List and View commands.

* Corrected repoUrl

* Made changes to follow Joshua's implementations. Contains vaguely-implemented methods.

* Changed parameters from EntityList to Model. Made some changes to follow Joshua's implementations.

* Moved ENTITY_TYPE to a static constant stored in CliSyntax.

* Updated code to use viewMinimal() and viewDetailed().

* Minor changes

* Implemented potential ways of executing each Participant command.

* Merge from upstream

* Updated code to use Model#updateParticipant's return value.

* Removed Commands associated with Issues.

* Removed Commands associated with Issues.

(cherry picked from commit cef69759ba683ab153084b6e274ea11472a12365)

* Imported Entities and other objects.

* Updated AboutUs.adoc and ContactUs.adoc

* Update UserGuide Introduction and Quickstart

* Update UserGuide Section 4

* Move generate id to readable entityList so that controller and parser can call it (#31)

* Alfred Add Command Parsers (#32)

* Add Alfred-specific parser classes.

* Modify prefixes.

* Implement Add Command Parsers.

* Made changes to teammate names.

* Made changes to teammate names.

(cherry picked from commit 7c8d00aa67de6068d6176dbee5b0dcd9dbe91055)

* Fix Storage Compilation Errors (#42)

* Update build.gradle file (minor changes)

* Complete set-up for project (as per the Developer Guide)

* Update README.adoc with Travis CI build status

* Update some files to ensure that ./config/travis/run-checks.sh runs fine

* Create the 4 different ListStorages. Create AlfredStorage. Create AlfredStorageManager (halfway done).

* Creation of all necessary Storage classes (with the exception of the TeamListStorage handler classes)

* Added Storage classes for TeamList

* Fix bracket issue

* Update Storage classes with appropriate JSON adaptation

* Update User Guide

* Fixed all compilation problems

* Alfred Delete Command Parser (#33)

* Add Alfred-specific parser classes.

* Modify prefixes.

* Implement Add Command Parsers.

* Implement Alfed delete command parsers.

* Standardise command line syntax.

* Revert "Alfred Delete Command Parser (#33)" (#43)

This reverts commit 2da432d0e9545d26af46c9bfe678fa65fb36eaba.

* Update gitignore to ignore binary files (#44)

* Fix compileTestJava (#45)

* Update build.gradle file (minor changes)

* Complete set-up for project (as per the Developer Guide)

* Update README.adoc with Travis CI build status

* Update some files to ensure that ./config/travis/run-checks.sh runs fine

* Create the 4 different ListStorages. Create AlfredStorage. Create AlfredStorageManager (halfway done).

* Creation of all necessary Storage classes (with the exception of the TeamListStorage handler classes)

* Added Storage classes for TeamList

* Fix bracket issue

* Update Storage classes with appropriate JSON adaptation

* Update User Guide

* Fixed all compilation problems

* Stop-gap fix for compileTestJava

* Stop-gap measures to fix compileTestJava while fixing codebase

* Fix add participant parser (#47)

* Changed code to follow checkstyle 8.1

* Checkstyle: Commands and Parsers (#51)

* Changed code to follow checkstyle

* Checkstyle EntityList (#53)

* Updated checkstyle

* Checkstyle: SubjectName.java (#54)

* Feat/checkstyle (#55)

* Feat/checkstyle (#56)

* Implement Basic Storage Tests and Fix Compile Errors (#57)

* Update build.gradle file (minor changes)

* Complete set-up for project (as per the Developer Guide)

* Update README.adoc with Travis CI build status

* Update some files to ensure that ./config/travis/run-checks.sh runs fine

* Create the 4 different ListStorages. Create AlfredStorage. Create AlfredStorageManager (halfway done).

* Creation of all necessary Storage classes (with the exception of the TeamListStorage handler classes)

* Added Storage classes for TeamList

* Fix bracket issue

* Update Storage classes with appropriate JSON adaptation

* Update User Guide

* Fixed all compilation problems

* Stop-gap fix for compileTestJava

* Stop-gap measures to fix compileTestJava while fixing codebase

* Ensure that ParticipantList can be saved and read from file

* Remove all dependencies on Issues and IssueList for Storage

* Create TypicalMentors.java file

* Update SubjectName enum values

* Update TypicalMentors.java

* Fix compileTestJava compilation errors

* Fix remaining checkstyle errors (#58)

* Modify add command parsers to handle missing arguments.

* Fix checkstyle errors.

* Finished implementation of Commands for Team and Mentor

* Changed code to follow checkstyle

* Placeholders for each team member's adoc

* Edit add parser arguments to make them consistent.

* Rename addressbookparser to AlfredParser.

* Edited code to reflect feedback

* Added COMMAND_USAGE to all commands

* Possible method to add participant/mentor to teams

* Implemented test for AddParticipantCommand and StringUtil class

* Fixed checkstyle for tests

* Fixed checkstyle issues

* Fixed checkstyle issues

(cherry picked from commit 6dce6643d1e6687d258294146cf91805b509e670)

* Implemented LoadCommand

* Implemented ExportCommand

* Fixed issues with Export and Load Command

* Updated ListCommands

* Implemented test for AddParticipantCommand and StringUtil class

* Fixed checkstyle for tests

* Fixed issues with AddParticipantCommandTest

* Implemented AddCommandTests for all Entities

* Implemented DeleteCommandTest for all entities and repackaged tests

* Implemented ListCommandTests for all entities

* Implemented ViewCommandTests for all Entities

* Implemented test for AddParticipantCommand and StringUtil class

* Fixed checkstyle for tests

* Implemented test for AddParticipantCommand and StringUtil class

* Fixed issues with AddParticipantCommandTest

* Implemented AddCommandTests for all Entities

* Implemented DeleteCommandTest for all entities and repackaged tests

* Implemented ListCommandTests for all entities

* Implemented ViewCommandTests for all Entities

* Implemented EditCommandTests and EditDescriptorTests for all entities and updated other tests

* Removed potential issue of travis failure

* Implemented ExportCommand for all entities and updated the contains() method in all implementations of EntityList

* Implemented Parsers for Export and Load Commands

* Implemented LoadCommand

* Implemented ExportCommand

* Fixed issues with Export and Load Command

* Implemented ExportCommand for all entities and updated the contains() method in all implementations of EntityList

* Implemented Parsers for Export and Load Commands

* Updated Export and Load Commands

* Fixed checkstyle issues and refactored LoadCommand

* Updated validation of SubjectName and ProjectType and other minor updates

* Fixed checkstyle issue

* Implemented LoadCommand

* Implemented ExportCommand

* Fixed issues with Export and Load Command

* Implemented ExportCommand for all entities and updated the contains() method in all implementations of EntityList

* Implemented Parsers for Export and Load Commands

* Implemented LoadCommand

* Implemented ExportCommand

* Fixed issues with Export and Load Command

* Implemented ExportCommand for all entities and updated the contains() method in all implementations of EntityList

* Implemented Parsers for Export and Load Commands

* Updated Export and Load Commands

* Fixed checkstyle issues and refactored LoadCommand

* FIxed error

* Implemented tests for CsvUtil

* Implemented tests for Id

* Implemented tests for Error and ErrorTracker

* Implemented tests for LoadCommand

* Fixed checkstyle issues

* Implemented tests for LoadCommandParser

* Implemented tests for ExportCommandParser

* Implemented tests for ExportCommand

* Implemented tests for all subclasses of ExportCommands

* Fixed checkstyle issues

* Updated CommandException

* Updated LoadCommand and ExportCommand to allow loading/exporting of Participants and Mentors of Teams through csv

* Updated tests for Load and Export Commands

* Fixed checkstyle issues

* Fixed travis issues

* Fixed potential Travis issues

* Fixed travis

* Fixed checkstyle issues

* Updated code to reflect Abhiman's suggestions

* Updated user guide to fit Export and Import Command

* Updated code to reflect Joshua's comments

* Fixed checkstyle issues

* Fixed failing tests

* Changed method name

* Updated user guide to fit Export and Import Command

* Updated ID regex and loading of entities

* Updated ViewCommand and refactored listcommand

* Fixed tests

* Fixed checkstyle issues

* Some changes on UI code

* Changed IOException message

* Add AlfredParserUtil Tests.

* Remove project type field.

* Updated ImportCommand to create an output csv file with errors if an exception is thrown while importing data

* Deprecated ProjectType

* Updated ImportCommand to create errorFile only if file path is specified

* Add tests for AlfredParser.

* Updated tests for ImportCommand and ImportCommandParser

* Fix checkstyle errors.

* Fixed checkstyle issues

* save work.

* Add more test cases for AlfredParser.

* Fix disabled tests to work.

* Add code for edit command.

* Fixed PR issue

* Add edit command parsers.

* Fix checkstyle errors.

* Corrected tests

* Merged feat/deprecate-projecttype

* Removed csv file

* Deleted placeholders for assign commands

* Fixed issues with deletion

* Fixed tests

* Fixed checkstyle issues

* Created tests for DeleteMentorCommand

* Removed @disabled for one test

* Fixed travis issues

* fixed checkstyle

* Updated to enable jar files

* Added ImportCommand details in DG

* Updated EditCommand

john0227 committed:
Revert "EditCommand (#310)" (#311)

hcwong committed:
Add codecov yml to ignore ui (#309)

john0227 committed:
Updating DG (#313)

hcwong committed:
Deprecate Ab3 model 2 (#169)

hcwong committed:
Make find case insenstitive (#290)

Abhiman2211 committed:
Ugfixpt2 (#314)

justarock111 committed:
Update command navigation events (#315)

brianyenna committed:
Update user guide v1.4 (#317)

* Update User Guide for command history navigation, undo and redo

* Updat User Guide

john0227 committed:
EditCommand (#312)

Abhiman2211 committed:
Scoringdevguide (#318)

hcwong committed:
Revert "Make find case insenstitive (#290)" (#327)

justarock111 committed:
Update AutoCompleteCommandBox and edited event handler (#330)

hcwong committed:
Refactored the initialize method to be less ugly (#129)

hcwong committed:
Model Manager Tests + Delete Persons PR. (#316)

brianyenna committed:
Create final tests (#340)

* Create CommandRecordTest and CommandHistoryManagerTest

* Created equality tests for UndoCommand and RedoCommand

hcwong committed:
Fix typos in find that are causing bugs (#328)

hcwong committed:
Update get commands for model (#331)

brianyenna committed:
Update user guide (Undo/Redo/History) (#341)

Abhiman2211 committed:
General Improvements for Code (#329)

Abhiman2211 committed:
Ugupdate (#333)

brianyenna committed:
Update developer guide and user guide (#346)

Abhiman2211 committed:
Finalise Scoring Section in Dev Guide (#345)

brianyenna committed:
Near last fixes (#337)

hcwong committed:
Close help window once other command is typed in (#343)

hcwong committed:
Fix Find case insensitivity (#342)

hcwong committed:
Fix Find Userguide (#348)

justarock111 committed:
fix bug (#332)

john0227 committed:
Fix/import export (#350)

john0227 committed:
Revert "Fix/import export (#350)" (#351)

john0227 committed:
Fix tilde in file path (#353)

brianyenna committed:
Update Home and Help commands for Command History Navigation Feature (#355)

justarock111 committed:
Create tests for assign/remove command (#347)

brianyenna committed:
Update and Finalise PPP (#358)

* Update PPP

* update with attempt to fix include bug

* Create PPP pdf

* Update PPP and generated PDF

* Finalise PPP

hcwong committed:
Update find and model DG (#359)

john0227 committed:
UGDG Update (#349)

john0227 committed:
update ppp (#360)

john0227 committed:
Update johnmin.adoc

john0227 committed:
Update johnmin.adoc

john0227 committed:
Update johnmin.adoc

hcwong committed:
PPP Joshua (#362)

Abhiman2211 committed:
Add changes to UG, DG and PPP (#363)

brianyenna committed:
Update ppp (#365)

justarock111 committed:
Update user guide and developer guide and project portfolio (#361)

brianyenna committed:
Standardise dg (#367)

* Update PPP

* Update DG for Design Considerations Formatting Standardisation

brianyenna committed:
Standardise dg (#369)

hcwong committed:
Fix typos for Joshua's stuff (#366)

Abhiman2211 committed:
Edit to Alfred Documentation (#368)

john0227 committed:
Update PPP (#364)

hcwong committed:
Add logic description (#370)

justarock111 committed:
Fix bug in command suggestion box (#371)

Abhiman2211 committed:
Miscellaneous Changes to Help Command and DG (#375)

john0227 committed:
Update Error Message ImportCommand (#376)

john0227 committed:
Update UserGuide.adoc

justarock111 committed:
fix bug  update ugdg (#377)

hcwong committed:
Fix delete participant bug (#373)

hcwong committed:
Update find suggestion templates (#372)

justarock111 committed:
Feat/fix bug 5.0 (#378)

Abhiman2211 committed:
Merge PPP Changes into Master (#379)

hcwong committed:
Fix filenames for PPP (#388)
